google_guest_agent/oslogin.go

```diff
@@ -340,35 +350,28 @@ func createOSLoginSudoersFile() error {
 	return sudoFile.Close()
 }
 
-// restartService tries to restart a service on linux-like systems. It attempts
-// to find and use the following mechanisms in order:
-// 1. The `systemctl` utility, if in a systemd environment.
-// 2. The `service` command, if present.
-// 3. A SysVinit script directly, if present.
-// Missing mechanisms and missing or disabled services are ignored.
+// restartService tries to restart a systemd service if it is already running.
 func restartService(servicename string) error {
-	init, err := os.Readlink("/sbin/init")
-	if err == nil && strings.Contains(init, "systemd") {
-		if systemctl, err := exec.LookPath("systemctl"); err == nil {
-			if err := runCmd(exec.Command(systemctl, "is-active", servicename+".service")); err == nil {
-				return runCmd(exec.Command(systemctl, "restart", servicename+".service"))
-			}
-			return nil
+	if systemctl, err := exec.LookPath("systemctl"); err == nil {
+		if err := runCmd(exec.Command(systemctl, "is-active", servicename+".service")); err == nil {
+			return runCmd(exec.Command(systemctl, "restart", servicename+".service"))
 		}
 	}
-	service, err := exec.LookPath("service")
-	if err == nil {
-		if err := runCmd(exec.Command(service, servicename, "status")); err == nil {
-			return runCmd(exec.Command(service, servicename, "restart"))
+
+	return nil
+}
+
+// startService tries to start a systemd service. If the service is already
+// running and restart is true, the service is restarted.
+func startService(servicename string, restart bool) error {
+	if systemctl, err := exec.LookPath("systemctl"); err == nil {
+		started := nil == runCmd(exec.Command(systemctl, "is-active", servicename+".service"))
+		if !started {
+			return runCmd(exec.Command(systemctl, "start", servicename+".service"))
 		}
-		return nil
-	}
-	initService := "/etc/init.d/" + servicename
-	if _, err := os.Stat(initService); err == nil {
-		if err := runCmd(exec.Command(initService, "status")); err == nil {
-			return runCmd(exec.Command(initService, "restart"))
+		if started && restart {
+			return runCmd(exec.Command(systemctl, "restart", servicename+".service"))
```

Comment:
Why not just call restartService(servicename)- at least this would make a bit more sense to me than dup'ing the exec call.

Reply:
that would dupe the is-active call. i thought this was more readable and mapped to logical behaviors.